src/views/catalog/CreateFromInstanceTypes/CreateFromInstanceType.tsx

```diff
@@ -1,80 +1,29 @@
-import { getInstanceTypeFromVolume } from './components/AddBootableVolumeModal/utils/utils';
 import BootableVolumeList from './components/BootableVolumeList/BootableVolumeList';
 import CreateVMFooter from './components/CreateVMFooter/CreateVMFooter';
 import SectionListItem from './components/SectionListItem/SectionListItem';
-import useBootableVolumes from './hooks/useBootableVolumes';
-import useInstanceTypesAndPreferences from './hooks/useInstanceTypesAndPreferences';
-import {
-  BootableVolume,
-  initialInstanceTypeState,
-  INSTANCE_TYPES_SECTIONS,
-  InstanceTypeState,
-} from './utils/constants';
-import { generateVM, getBootableVolumePVCSource } from './utils/utils';
+import { useInstanceTypeVMStore } from './state/useInstanceTypeVMStore';
+import { INSTANCE_TYPES_SECTIONS } from './utils/constants';
 
 import './CreateFromInstanceType.scss';
 
 const CreateFromInstanceType: FC = () => {
-  const [ns] = useActiveNamespace();
-  const history = useHistory();
-
-  const { pvcSources, ...BootableVolumesObject } = useBootableVolumes();
-  const { preferences, loadError } = useInstanceTypesAndPreferences();
-  const preferencesMap = useMemo(() => convertResourceArrayToMap(preferences), [preferences]);
-
+  const { t } = useKubevirtTranslation();
   const sectionState = useState<INSTANCE_TYPES_SECTIONS>(INSTANCE_TYPES_SECTIONS.SELECT_VOLUME);
 
-  const [selectedBootableVolume, setSelectedBootableVolume] = useState<BootableVolume>();
-  const [selectedInstanceType, setSelectedInstanceType] =
-    useState<InstanceTypeState>(initialInstanceTypeState);
-  const [vmName, setVMName] = useState<string>(
-    uniqueNamesGenerator({
-      dictionaries: [adjectives, animals],
-      separator: '-',
-    }),
-  );
-  const [sshSecretCredentials, setSSHSecretCredentials] = useState<SSHSecretCredentials>({
-    sshSecretName: '',
-    sshSecretKey: '',
-  });
-  const [pvcSource, setPVCSource] = useState<IoK8sApiCoreV1PersistentVolumeClaim>();
-
-  const namespace = ns === ALL_NAMESPACES_SESSION_KEY ? DEFAULT_NAMESPACE : ns;
-
-  const onSelectVolume = useCallback(
-    (selectedVolume: BootableVolume) => {
-      const instanceType = getInstanceTypeFromVolume(selectedVolume);
-
-      setSelectedBootableVolume(selectedVolume);
-      setPVCSource(getBootableVolumePVCSource(selectedVolume, pvcSources) ?? null);
-      setSelectedInstanceType(instanceType);
-    },
-    [pvcSources],
-  );
+  const { resetInstanceTypeVMState } = useInstanceTypeVMStore();
 
-  useEffect(() => {
-    if (!isEmpty(selectedBootableVolume) && isEmpty(pvcSource)) {
-      setPVCSource(getBootableVolumePVCSource(selectedBootableVolume, pvcSources) ?? null);
-    }
-  }, [pvcSource, pvcSources, selectedBootableVolume]);
+  // in case of starting this form and than moving to different nav item
+  // we want a clean form so we run this on initial render
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
?

Reply:
should I include the reset function as a dep?